docs/diagrams/CourseValidator.puml

```diff
@@ -0,0 +1,45 @@
+@startuml
+hide footbox
+!include .SequenceDiagram.config
+participant ":AddCoursesCommand" as AddCoursesCommand
+participant ":CourseValidator" as CourseValidator
+
+activate AddCoursesCommand
+
+AddCoursesCommand -> CourseValidator:isValidInput(nusCourseInput: String, pu: String, puCourseInput: String, jsonObject: JsonObject)
+activate CourseValidator
+
+CourseValidator -> CourseValidator:getPUCourseList
+activate CourseValidator
+
+alt matchPu != null
+CourseValidator -> CourseValidator: get JsonArray for Partner University
+else else
+CourseValidator -> CourseValidator:displayPartnerUniversities()
+activate CourseValidator
+deactivate
+CourseValidator -> CourseValidator: throw IllegalArgumentException
+end
+return courses: JsonArray
+
+
+CourseValidator -> CourseValidator: isValidCourseMapping
```

Comment:
![image](https://github.com/user-attachments/assets/bcc15ccf-414d-47fc-9059-cbb509ed51be)
![image](https://github.com/user-attachments/assets/ce36b061-04b8-4ea2-aa5d-38cef04dd582)
throw: though there is no standard notation, it may be better to show the throw as a call to exception object rather than self invocation

Reply:
![image](https://github.com/user-attachments/assets/5519c7e4-3ae9-40f7-992a-3901bde77bf9)

I think class diagram cannot use this kind of arrows.(above) and i don't think it is supposed to be -->?

![image](https://github.com/user-attachments/assets/c751bdde-9d5c-471d-95ae-e787ffd3f14b)

Reply:
Thanks for pointing it out! I have made the changes in the next few commits!